spec/requests/articles_controller_spec.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ArticlesController, type: :controller do
+  context 'articles user not authenticated' do
+    describe '#index' do
+      before { get :index }
+
+      it { should respond_with 200 }
+      it { should render_template 'index' }
+    end
+
+    describe '#show' do
+      subject { create :article }
+
+      it 'article found' do
+        get :show, params: { id: subject.id }
+        expect(response.status).to eq(200)
+      end
+
+      it 'article not found' do
+        random_id = rand(1..100)
+        expect { get get :show, params: { id: random_id } }.to raise_error ActionController::RoutingError
+      end
+    end
+
+    describe '#new' do
+      before { get :new }
+
+      it { should respond_with(:redirect) }
+      it { should redirect_to signin_path }
+    end
+
+    describe '#create' do
+      before { post :create }
+
+      it { should respond_with(:redirect) }
+      it { should redirect_to signin_path }
+    end
+
+    describe '#update' do
+      let(:article) { create :article }
+
+      before { get :create, params: { id: article.id } }
+      it { should respond_with(:redirect) }
+      it { should redirect_to signin_path }
+    end
+
+    describe '#destroy' do
+      let(:article) { create :article }
+
+      before { get :destroy, params: { id: article.id } }
+      it { should respond_with(:redirect) }
+      it { should redirect_to signin_path }
+    end
+  end
+end
+
+RSpec.describe ArticlesController, type: :request do
```

Comment:
Why are you testing the actions twice? You have `controller` and `request` tests.

Reply:
I testing the Controller as `controller` type, and `request` type because with the `request` type I can log in with the user. Should I use only one type ?

Reply:
Yes, please use `request` type.